.github/ci/override_managed_component.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python
+#
+# SPDX-FileCopyrightText: 2024 Espressif Systems (Shanghai) CO LTD
+# SPDX-License-Identifier: Apache-2.0
+
+import sys
+import argparse
+from pathlib import Path
+from glob import glob
+from idf_component_tools.manifest import ManifestManager
+
+
+def override_with_local_component(component, local_path, app):
+    app_path = Path(app)
+
+    absolute_local_path = Path(local_path).absolute()
+    if not absolute_local_path.exists():
+        print('[Error] {} path does not exist'.format(local_path))
+        raise Exception
+    if not app_path.exists():
+        print('[Error] {} path does not exist'.format(app_path))
+        raise Exception
+
+    print('[Info] Processing app {}'.format(app))
+    manager = ManifestManager(app_path / 'main', 'app')
```

Comment:
Would it be safer if argparse asked for the path to the app's `idf_component.yml` directly. We currently assume that for all tinyusb device examples, their dependencies for the tinyusb component is specified in the main component's manifest file.

But it might be possible in the future where some example defines a local component X, and it is component X's manifest file which declares the dependency on tinyusb.

Reply:
Yes, good idea.
Currently, I don't think I can do that fast, so I create a jira ticket and return back to that later.